frontend/www/js/omegaup/arena/location.ts

```diff
@@ -63,3 +66,50 @@ export function getOptionsFromLocation(location: string): LocationOptions {
 
   return response;
 }
+
+export async function getProblemAndRunDetails({
+  location,
+  problems,
+}: {
+  location: string;
+  problems?: types.NavbarProblemsetProblem[];
+}): Promise<{
+  runDetails: null | types.RunDetails;
+  problemDetails: null | types.ProblemDetails;
+}> {
+  const { guid, problemAlias } = getOptionsFromLocation(location);
+  let runDetails: null | types.RunDetails = null;
+  let problemDetails: null | types.ProblemDetails = null;
+  let problemPromise: Promise<null | types.ProblemDetails> = Promise.resolve(
+    null,
+  );
+  let runPromise: Promise<null | types.RunDetails> = Promise.resolve(null);
+
+  if (problemAlias) {
+    problemPromise = api.Problem.details({
+      problem_alias: problemAlias,
+      prevent_problemset_open: false,
+    });
+  }
+  if (guid) {
+    runPromise = api.Run.details({ run_alias: guid });
+  }
+
+  [problemDetails, runDetails] = await Promise.all([
```

Comment:
```suggestion
  const [problemDetails, runDetails] = await Promise.all([
```

hay que evitar el estilo de declarar cosas arriba y luego utilizarlas acá abajo: siempre hay que intentar declarar las cosas _justo_ antes de la primera vez que se usan.